### PR TITLE
Added quality declaration

### PR DESCRIPTION
**Pull request by ahcorde**
Added quality declaration

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+This document is a declaration of software quality for the `libstatistics_collector` package, based on the guidelines in [REP-2004](https://github.com/ros-infrastructure/rep/blob/rep-2004/rep-2004.rst).
```
Comment (dabonnie) [inline QUALITY_DECLARATION.md:1]: The REP-2004 link is broken, expected?

File: QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+### Public API Documentation [3.ii]
+
+The API is not publicly available yet.
+
+### License [3.iii]
+
+The license for `libstatistics_collector` is Apache 2.0, and a summary is in each source file, the type is declared in the [`package.xml`](./package.xml) manifest file, and a full copy of the license is in the [`LICENSE`](LICENSE) file.
+
+There is an automated test which runs a linter that ensures each file has a license statement. [Here](http://build.ros2.org/view/Epr/job/Epr__libstatistics_collector__ubuntu_bionic_amd64/lastCompletedBuild/testReport/libstatistics_collector/) can be found a list with the latest results of the various linters being run on the package.
```
Comment (dabonnie) [inline QUALITY_DECLARATION.md:83]: The linter test link is broken, expected?

File: QUALITY_DECLARATION.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+### Version Stability [1.ii]
+
+`libstatistics_collector` is at a stable version (`>= 1.0.0`.)
+The current version can be found in its [package.xml](package.xml), and its change history can be found in its [CHANGELOG](CHANGELOG.rst).
+
+### Public API Declaration [1.iii]
+
+All symbols in the installed headers are considered part of the public API.
+
+Except for the exclusions listed below, all installed headers are in the `include` directory of the package, headers in any other folders are not installed and considered private.
```
Comment (brawner) [inline QUALITY_DECLARATION.md:24]: There are no exclusions.

File: QUALITY_DECLARATION.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+### API Stability Policy [1.iv]
+
+`libstatistics_collector` will not break public API within a released ROS distribution, i.e. no major releases once the ROS distribution is released.
```
Comment (brawner) [inline QUALITY_DECLARATION.md:28]: This section is supposed to talk about API stability within major versions right?

File: QUALITY_DECLARATION.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+### ABI Stability Policy [1.v]
+
+`libstatistics_collector` contains C++ code and therefore must be concerned with ABI stability, and will maintain ABI stability within a ROS distribution.
```
Comment (brawner) [inline QUALITY_DECLARATION.md:32]: Same

File: QUALITY_DECLARATION.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+`libstatistics_collector` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process).
+
+### Change Requests [2.i]
+
+All changes will occur through a pull request, check [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change-control-process) for additional information.
+
+### Contributor Origin [2.ii]
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](./CONTRIBUTING.md).
```
Comment (brawner) [inline QUALITY_DECLARATION.md:48]: Should there be a signoff clause in the contributing.md for it to be DCO?

File: QUALITY_DECLARATION.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`libstatistics_collector` has a [feature list](https://github.com/ros-tooling/libstatistics_collector/blob/master/README.md) and each item in the list links to the corresponding feature documentation. There is documentation for all of the features, and new features require documentation before being added.
+
+### Public API Documentation [3.ii]
+
+The API is not publicly available yet.
```
Comment (brawner) [inline QUALITY_DECLARATION.md:77]:
```suggestion
The API documentation is not publicly available yet.

```

File: QUALITY_DECLARATION.md (hunk 2 above)
Comment (brawner) [reply QUALITY_DECLARATION.md:83]: Probably should point to ci.ros2.org. @ahcorde Fix this link and below.

https://ci.ros2.org/job/nightly_linux_release/lastBuild/testReport/libstatistics_collector/

File: QUALITY_DECLARATION.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+The `std_msgs` package provides many basic message types.
+
+It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/common_interfaces/blob/master/std_msgs/QUALITY_DECLARATION.md).
+
+It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
+It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
+
+### Direct Runtime non-ROS Dependency [5.iii]
+
+`libstatistics_collector` has no run-time or build-time dependencies that need to be considered for this declaration.
```
Comment (brawner) [inline QUALITY_DECLARATION.md:179]:
```suggestion
`libstatistics_collector` has no non-ROS runtime or build-time dependencies that need to be considered for this declaration.

```

File: QUALITY_DECLARATION.md (hunk 1 above)
Comment (brawner) [reply QUALITY_DECLARATION.md:1]: This is expected, eventually the rep-2004 PR will be merged...

File: QUALITY_DECLARATION.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+- achieving and maintaining a reasonable branch line coverage (90-100%)
+- no lines are manually skipped in coverage calculations
+
+Changes are required to make a best effort to keep or increase coverage before being accepted, but decreases are allowed if properly justified and accepted by reviewers.
+
+Each time that a new PR is created [codecov](https://codecov.io/) will create diagram to show the current status of the test coverage and how this PR will affect the coverage of the package. Latest codecov result report can be seen [here](https://codecov.io/gh/ros-tooling/libstatistics_collector).
+
+### Performance [4.iv]
+
+It is not yet defined if this package requires performance testing and how addresses this topic.
```
Comment (brawner) [inline QUALITY_DECLARATION.md:126]: "This package does not yet have performance testing."

Comment (dabonnie) [reply QUALITY_DECLARATION.md:126]: Is this created manually for each package or is there template defined somewhere?

File: QUALITY_DECLARATION.md (hunk 1 above)
Comment (chapulina) [reply QUALITY_DECLARATION.md:1]: I think we should use the published link: https://www.ros.org/reps/rep-2004.html

File: QUALITY_DECLARATION.md (hunk 9 above)
Comment (ahcorde) [reply QUALITY_DECLARATION.md:126]: Yes, it's created manually

File: QUALITY_DECLARATION.md (hunk 4 above)
Comment (brawner) [reply QUALITY_DECLARATION.md:28]: This still needs to be addressed.

File: QUALITY_DECLARATION.md (hunk 9 above)
Comment (brawner) [reply QUALITY_DECLARATION.md:126]: Sentence still needs to be fixed.

**Comment by chapulina** [comment]
@dabonnie , is there anything missing to get this in?

**Pull request merged**